src/OnlineRetailPortal.Web/Translators/GetCategoriesWebResponse.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OnlineRetailPortal.Web
+{
+    public static class ServiceResponseTranslator
+    {
+        public static CategoryResponse ToCategoriesContract(this Contracts.GetCategoriesServiceResponse getCategoriesServiceResponse)
+        {
+            CategoryResponse response = new CategoryResponse()
+            {
+                Categories = getCategoriesServiceResponse.Categories.Select(x => new Category()
```

Comment:
we might need a null check here for categories

Reply:
And for getCategoriesServiceResponse as well

Reply:
done